src/main/java/seedu/address/model/job/Money.java

```diff
@@ -0,0 +1,136 @@
+package seedu.address.model.job;
+
+import static java.util.Objects.requireNonNull;
+
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.text.NumberFormat;
+import java.util.Locale;
+
+/**
+ * Represents some value of money. Immutable.
+ */
+public class Money {
+
+    private static final int VALUE_SCALE = 6;
+    private static final NumberFormat CURRENCY_FORMAT = NumberFormat.getCurrencyInstance(Locale.US);
+
+    public final BigDecimal value;
+
+    /**
+     * Constructs a {@code Money}.
+     *
+     * @param value A value as a double.
+     */
+    public Money(double value) {
+        this(BigDecimal.valueOf(value).setScale(VALUE_SCALE, RoundingMode.HALF_UP));
+    }
+
+    /**
+     * Constructs a {@code Money}.
+     *
+     * @param value A value as a BigDecimal.
+     */
+    public Money(BigDecimal value) {
```

Comment:
Perhaps this can be private instead. Or is it public because it is easier to test?